test/deprecated-keyword/messageFiltering.chpl

```diff
@@ -0,0 +1,228 @@
+var x1001 = 1001;
+deprecated "Lorem ipsum dolor sit amet"
+var x1002 = 1002;
+deprecated "Lorem_ipsum_dolor_sit_amet"
+var x1003 = 1003;
+deprecated "mod proc iter data const var param type class record"
+var x1004 = 1004;
+deprecated ":mod proc:`abc` :iter data:`def` :const var:`ghi`"
+var x1005 = 1005;
+deprecated ":param type:`jkl` :class record:`mno`"
+var x1006 = 1006;
+
+// Test filtering at start, middle, and end of msg (with and without content):
+deprecated "--- Test filtering at start, middle, and end of msg (with and without content): ---"
+var x2000 = 2000;
+deprecated ":proc:`abc` Lorem ipsum dolor sit amet"
+var x2001 = 2001;
+deprecated "Lorem ipsum :proc:`abc` dolor sit amet"
+var x2002 = 2002;
+deprecated "Lorem ipsum dolor sit amet :proc:`abc`"
+var x2003 = 2003;
+deprecated ":proc:`abc` Lorem :proc:`def` ipsum dolor sit amet :proc:`ghi`"
+var x2004 = 2004;
+
+// Test different text in ::s (all should resolve)
+deprecated "--- Test different text in ::s (all should resolve) ---"
+var x3000 = 3000;
+deprecated "Lorem ipsum :mod:`test` dolor sit amet"
+var x3001 = 3001;
+deprecated "Lorem ipsum :proc:`test` dolor sit amet"
+var x3002 = 3002;
+deprecated "Lorem ipsum :iter:`test` dolor sit amet"
+var x3003 = 3003;
+deprecated "Lorem ipsum :data:`test` dolor sit amet"
+var x3004 = 3004;
+deprecated "Lorem ipsum :const:`test` dolor sit amet"
+var x3005 = 3005;
+deprecated "Lorem ipsum :var:`test` dolor sit amet"
+var x3006 = 3006;
+deprecated "Lorem ipsum :param:`test` dolor sit amet"
+var x3007 = 3007;
+deprecated "Lorem ipsum :type:`test` dolor sit amet"
+var x3008 = 3008;
+deprecated "Lorem ipsum :class:`test` dolor sit amet"
+var x3009 = 3009;
+deprecated "Lorem ipsum :record:`test` dolor sit amet"
+var x3010 = 3010;
+deprecated "Lorem ipsum :attr:`test` dolor sit amet"
+var x3011 = 3011;
+deprecated "Lorem :mod:`abc` ipsum :proc:`def` dolor :iter:`ghi` sit :data:`jkl` amet"
+var x3012 = 3012;
+deprecated "Lorem :const:`abc` ipsum :var:`def` dolor :param:`ghi` sit :type:`jkl` amet"
+var x3013 = 3013;
+deprecated "Lorem :class:`abc` ipsum :record:`def` dolor :attr:`ghi` sit amet"
+var x3014 = 3014;
+
+// Test different text in ::s (all should not resolve)
+deprecated "--- Test different text in ::s (all should not resolve) ---"
+var x4000 = 4000;
+deprecated "Lorem ipsum ::`abc` dolor sit amet"
+var x4001 = 4001;
+deprecated "Lorem ipsum :a:`abc` dolor sit amet"
+var x4002 = 4002;
+deprecated "Lorem ipsum :abc123def:`abc` dolor sit amet"
+var x4003 = 4003;
+deprecated "Lorem ipsum :123abc123def:`abc` dolor sit amet"
+var x4004 = 4004;
+deprecated "Lorem ipsum :abc123def123:`abc` dolor sit amet"
+var x4005 = 4005;
+deprecated "Lorem ipsum :123abc123def123:`abc` dolor sit amet"
+var x4006 = 4006;
+deprecated "Lorem ipsum :abc_123_def:`abc` dolor sit amet"
+var x4007 = 4007;
+
+// Test different text in ``s
+deprecated "--- Test different text in ``s ---"
+var x5000 = 5000;
+deprecated "Lorem ipsum :proc:`` dolor sit amet"
```

Comment:
Interesting that the `:proc:`` ` turns up for this one.  I think that's appropriate, but what happens if the only contents is a space?

Reply:
I just took another look at: https://chapel-lang.org/docs/latest/tools/chpldoc/chpldoc.html#inline-markup-2 and it looks like maybe you're thinking of the `:role:`title <target>`` syntax? I guess in this case we would want to filter out title and just emit target in the message.  Right now the regexp wouldn't match the space (it only matches alphanumeric chars plus underscores plus $s).

I want to get this particular PR submitted so its not blocking my other issue, but I think I'll fork off another issue to deal with these "additional facilities" as they're called on the doc page (see #18549). Probably something we can put a "Good First Issue" label on as well.

In the meantime I've also update the regexp to accept dots (.s) as that looks like another common use case.